pkg/ccl/changefeedccl/event_processing.go

```diff
@@ -99,11 +103,44 @@ func newEventConsumer(
 	enablePacer := changefeedbase.PerEventElasticCPUControlEnabled.Get(&cfg.Settings.SV)
 
 	makeConsumer := func(s EventSink, frontier frontier) (eventConsumer, error) {
-		var err error
+		var sourceNodeLocality, nodeName, nodeID string
+		tiers := cfg.Locality.Tiers
+
+		nodeLocalities := make([]string, 0, len(tiers))
+		for _, t := range tiers {
+			nodeLocalities = append(nodeLocalities, t.String())
+		}
+		sourceNodeLocality = strings.Join(nodeLocalities, ",")
+
+		nodeInfo := cfg.ExecutorConfig.(*sql.ExecutorConfig).NodeInfo
+		getPGURL := nodeInfo.PGURL
+		pgurl, err := getPGURL(url.User(username.RootUser))
+		if err != nil {
+			return nil, err
+		}
+		parsedUrl, err := url.Parse(pgurl.String())
+		if err != nil {
+			return nil, err
+		}
+		host, _, err := net.SplitHostPort(parsedUrl.Host)
+		if err == nil {
+			nodeName = host
+		}
+
+		if optionalNodeID, ok := nodeInfo.NodeID.OptionalNodeID(); ok {
+			nodeID = optionalNodeID.String()
+		}
+
```

Comment:
can you pull this out into a function that constructs the enrichedSourceData, either here or a literal constructor for that.